examples/asr/vad_infer.py

```diff
@@ -0,0 +1,151 @@
+       to speech/no-speech segment in start and end times format in that script.]
+   
+Usage:
+python vad_infer.py  --vad_model="MatchboxNet-VAD-3x2" --dataset=<FULL PATH OF MANIFEST TO BE PERFORMED INFERENCE ON> --out_dir='frame/demo' --time_length=0.63
+
+"""
+
+
+import json
+import os
+from argparse import ArgumentParser
+
+import torch
+
+from nemo.collections.asr.models import EncDecClassificationModel
+from nemo.utils import logging
+
+try:
+    from torch.cuda.amp import autocast
+except ImportError:
+    from contextlib import contextmanager
+
+    @contextmanager
+    def autocast(enabled=None):
+        yield
+
+
+device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+
+
+def main():
+    parser = ArgumentParser()
+    parser.add_argument(
+        "--vad_model", type=str, default="MatchboxNet-VAD-3x2", required=False, help="Pass: 'MatchboxNet-VAD-3x2'"
+    )
+    parser.add_argument("--dataset", type=str, required=True, help="path of json file of evaluation data")
+    parser.add_argument("--out_dir", type=str, default="vad_frame", help="dir of your vad outputs")
+    parser.add_argument("--time_length", type=float, default=0.63)
+    parser.add_argument("--shift_length", type=float, default=0.01)
+    args = parser.parse_args()
+
+    torch.set_grad_enabled(False)
+
+    if args.vad_model.endswith('.nemo'):
+        logging.info(f"Using local VAD model from {args.vad_model}")
+        vad_model = EncDecClassificationModel.restore_from(restore_path=args.vad_model)
+    else:
+        logging.info(f"Using NGC cloud VAD model {args.vad_model}")
+        vad_model = EncDecClassificationModel.from_pretrained(model_name=args.vad_model)
+
+    if not os.path.exists(args.out_dir):
+        os.mkdir(args.out_dir)
+
+    # setup_test_data
+    vad_model.setup_test_data(
+        test_data_config={
+            'vad_stream': True,
+            'sample_rate': 16000,
+            'manifest_filepath': args.dataset,
+            'labels': ['infer',],
+            'num_workers': 20,
+            'shuffle': False,
+            'time_length': args.time_length,
+            'shift_length': args.shift_length,
+            'trim_silence': False,
+        }
+    )
+
+    vad_model = vad_model.to(device)
+    vad_model.eval()
+
+    data = []
+    for line in open(args.dataset, 'r'):
+        sub_folder, file = (
+            json.loads(line)['audio_filepath'].split("/")[-2],
+            json.loads(line)['audio_filepath'].split("/")[-1],
+        )
+        data.append(sub_folder + "-" + file.split(".wav")[0])
+    print(f"Inference on {len(data)} audio files/json lines!")
+
+    time_unit = int(args.time_length / args.shift_length)
+    trunc = int(time_unit / 2)
+    trunc_l = time_unit - trunc
+    all_len = 0
+
+    for i, test_batch in enumerate(vad_model.test_dataloader()):
+        if i == 0:
+            status = 'start' if data[i] == data[i + 1] else 'single'
```

Comment:
could you please briefly explain why start, next, and end are being tracked?

Reply:
sure will explain in code

Reply:
No not in code, for me to understand.

Reply:
Let's discuss this via slack